Source/Icebreaker/Services/MatchingService.cs

```diff
@@ -185,12 +197,51 @@ private string GetChannelUserObjectId(ChannelAccount account)
             return JObject.FromObject(account).ToObject<TeamsChannelAccount>()?.AadObjectId;
         }
 
+        /// <summary>
+        /// Parse through a list of pairing information.
+        /// </summary>
+        /// <param name="pairHistory">List of pairing information.</param>
+        /// <returns>Returns pairs from the most recent pairing phase. This can be changed to include
+        /// pairings from less recent pairing phases by changing the `latestIteration` threshold.</returns>
+        private Tuple<Dictionary<string, HashSet<string>>, int> ParsePairHistory(IList<PairInfo> pairHistory)
```

Comment:
When returning mixed or complex type tuples it can be helpful to use [tuple field names](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples#tuple-field-names). This makes it more explicit what each element is.

This _might_ not be possible as I'm not certain if this project uses C# 7. 

Also it is better to use interfaces in the return types rather than concretions. (admittedly this file already isn't a great example)
E.G. 
```csharp
private (priorMappings: IDictionary<string, ISet<string>>, lastIteration: int) ParsePairHistory(IList<PairInfo> pairHistory)

Reply:
I'm not sure if this would work because simpler things like tuple unpacking don't seem to work either.

Reply:
You can update the language version in the csproj file to 8. The app services now support that.